src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/ConfirmationStatementService.java

```diff
@@ -86,6 +86,14 @@ public ConfirmationStatementService(CompanyProfileService companyProfileService,
             return ResponseEntity.badRequest().body(companyValidationResponse);
         }
 
+        var allEntries = confirmationStatementSubmissionsRepository.findAll();
```

Comment:
We should check if the transaction already has a confirmation statement resource rather than then search through all the confirmation statement submissions (there will be about 3-5 million per year). We have the Transaction passed in as a param to the method, we should be able to check the number of confirmation statement resources it has and if it already has one then return the error code. I would also move the check into it's own method to stop this method getting bigger.

Reply:
As Mark said, you'll want to iterate through all of transaction.data.resources
And check each of the resources "kind" field to see if it == "confirmation-statement"

Reply:
Thanks i knew there would be a better way to do this didn't release I could find the confirmation statement submission from the transaction. Mark when you say move the check into its own method you mean outside the `createConfirmationStatement` method correct ?